In [1]:
import sys
from pathlib import Path
import os
import json

sys.path.append(str(Path.cwd().parent.parent))
print(str(Path.cwd().parent.parent))
print("Current working directory: ", os.getcwd())

from ModelDevelopment.CentralPipeline import CentralPipeline
from ModelDevelopment.ImageBatchPipeline import ImageBatchPipeline
from DataProcessing.DataPreProcessing import DataPaths

c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition
Current working directory:  c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\ModelDevelopment\experiments


In [2]:
gt_test = DataPaths.TEST_DATA_GT.value
#Schema of gt_test: {"0": 4, "1": 93, "2": 25, "3": -1, "4": -1,

# load the json and compile a list of all tracklets with -1 as not_legible and all other tracklets as legible
with open(gt_test, 'r') as f:
    data = json.load(f)
    legible = []
    illegible = []
    
    for key, value in data.items():
        if value == -1:
            illegible.append(key)
        else:
            legible.append(key)
    
print("Legible tracklets: ", legible)
print("Illegible tracklets: ", illegible)

Legible tracklets:  ['0', '1', '2', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '26', '27', '28', '30', '31', '32', '33', '34', '35', '36', '37', '42', '43', '44', '45', '46', '48', '52', '54', '55', '56', '57', '58', '59', '60', '62', '63', '64', '65', '66', '67', '68', '69', '70', '72', '73', '77', '80', '83', '84', '86', '88', '91', '92', '93', '94', '95', '96', '98', '99', '100', '101', '102', '103', '105', '106', '107', '109', '111', '112', '113', '114', '115', '116', '120', '123', '125', '126', '127', '128', '131', '132', '133', '135', '136', '137', '138', '139', '140', '141', '142', '143', '144', '145', '146', '148', '149', '150', '151', '152', '153', '154', '156', '157', '159', '160', '161', '163', '166', '169', '170', '171', '173', '178', '179', '180', '181', '185', '186', '187', '188', '189', '190', '191', '192', '193', '196', '197', '198', '201', '202', '203', '204', '205', '206', '207', '208', '209', '210', '212', '213', '214', '219'

In [3]:
# # Delete soccer_ball.json files
# import os
# root_path = r"C:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\test\images"

# for tracklet in range(1, 1210):
#   path = os.path.join(root_path, str(tracklet))
  
#   # If there is a .soccer_ball.json file in there, delete it
#   if os.path.exists(os.path.join(path, "soccer_ball.json")):
#     os.remove(os.path.join(path, "soccer_ball.json"))

In [ ]:
pipeline = CentralPipeline(
  tracklets_to_process_override=[109, 88, 34, 151, 19, 68, 67, 128, 57, 178, 189, 196, 214, 242, 258, 271, 304, 327, 320, 352, 375, 378, 414, 416, 439, 425, 451, 453, 454, 466, 499, 511, 513, 526, 528, 567, 576, 594, 619, 623, 646, 687, 682, 709, 711, 717, 739, 758, 767, 795, 788, 840, 843, 832, 841, 852, 847, 851, 856, 850, 868, 866, 871, 901, 902, 904, 900, 907, 928, 939, 944, 961, 984, 983, 1005, 1039, 1064, 1063, 1077, 1091, 1101, 1137, 1152, 1153, 1160, 1167, 1157, 1165, 1206],
  #tracklets_to_process_override=[327, 711, 907, 1064], # These ones are now passing 
  #tracklets_to_process_override=illegible,
  #num_tracklets=1,
  #num_images_per_tracklet=50,
  input_data_path=DataPaths.TEST_DATA_DIR.value,
  output_processed_data_path=DataPaths.PROCESSED_DATA_OUTPUT_DIR_TEST.value,
  common_processed_data_dir=DataPaths.COMMON_PROCESSED_OUTPUT_DATA_TEST.value,
  gt_data_path=DataPaths.TEST_DATA_GT.value,
  display_transformed_image_sample=False, # NOTE: DO NOT USE. Code is parallelized so we cannot show images anymore. Code breaks, but first one will show if True.
  num_image_samples=1,
  use_cache=False, # Set to false to rebuild the cache
  suppress_logging=False,
  
  # --- PARALLELIZATION PARAMS --- These settings are optimal for an NVIDIA RTX 3070 Ti Laptop GPU.
  num_workers=2,            # CRITICAL optimisation param. Adjust accordingly. 62
  tracklet_batch_size=32,   # CRITICAL optimisation param. Adjust accordingly. 32
  image_batch_size=100,     # CRITICAL optimisation param. Adjust accordingly. 200
  num_threads_multiplier=4  # CRITICAL optimisation param. Adjust accordingly. 3. Interpretation: num_threads = num_workers * num_threads_multiplier
  )

2025-03-29 22:37:34 [INFO] DataPreProcessing initialized. Universe of available data paths:
2025-03-29 22:37:34 [INFO] ROOT_DATA_DIR: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted
2025-03-29 22:37:34 [INFO] TEST_DATA_GT: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\test\test_gt.json
2025-03-29 22:37:34 [INFO] TRAIN_DATA_GT: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\train\train_gt.json
2025-03-29 22:37:34 [INFO] TEST_DATA_DIR: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\test\images
2025-03-29 22:37:34 [INFO] TRAIN_DATA_DIR: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\train\images
2025-03-29 22:37:34 [INFO] CHALLENGE_DATA_DIR: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\challenge\images
2

In [5]:
pipeline.run_soccernet(
  run_soccer_ball_filter=True,
  generate_features=True,
  run_filter=True,
  run_legible=True,
  run_legible_eval=True,
  run_pose=False, ##
  run_crops=False,
  run_str=False,
  run_combine=False,
  run_eval=False)

2025-03-29 22:37:34 [INFO] Running the SoccerNet pipeline.
2025-03-29 22:37:34 [INFO] Tracklet override applied. Using provided tracklets: 3, 4, 20, 21, 22, 23, 24, 25, 29, 38, 39, 40, 41, 47, 49, 50, 51, 53, 61, 71, 74, 75, 76, 78, 79, 81, 82, 85, 87, 89, 90, 97, 104, 108, 110, 117, 118, 119, 121, 122, 124, 129, 130, 134, 147, 155, 158, 162, 164, 165, 167, 168, 172, 174, 175, 176, 177, 182, 183, 184, 194, 195, 199, 200, 211, 215, 216, 217, 218, 224, 225, 226, 230, 231, 233, 234, 235, 240, 241, 243, 255, 257, 259, 260, 262, 265, 267, 269, 270, 274, 276, 277, 279, 281, 282, 283, 284, 285, 286, 287, 288, 291, 292, 293, 294, 295, 297, 302, 303, 306, 307, 312, 317, 319, 324, 325, 326, 334, 335, 336, 337, 340, 342, 345, 351, 354, 356, 370, 372, 373, 374, 376, 379, 386, 388, 391, 392, 394, 395, 396, 400, 401, 402, 406, 408, 409, 415, 417, 419, 421, 431, 432, 433, 434, 437, 438, 441, 444, 447, 450, 459, 462, 463, 465, 467, 468, 469, 471, 472, 477, 482, 487, 504, 505, 507, 514, 520, 527, 529, 

  0%|          | 0/12 [00:00<?, ?it/s]

2025-03-29 22:37:34 [INFO] Tracklet batch size: 32
2025-03-29 22:37:34 [INFO] Image batch size: 100
2025-03-29 22:37:34 [INFO] Number of workers: 2
2025-03-29 22:37:34 [INFO] Number of threads created: 8
2025-03-29 22:37:34 [INFO] Using double parallelization: multiprocessing + CUDA batch processing.


Processing tracklets (CUDA + CPU):   0%|          | 0/32 [00:00<?, ?it/s]

2025-03-29 22:37:49 [INFO] Creating placeholder data files for Soccer Ball Filter.
2025-03-29 22:37:49 [INFO] Removed cached tracklet feature file (use_cache: False): c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\3\features.npy
2025-03-29 22:37:49 [INFO] Removed cached tracklet feature file (use_cache: False): c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\4\features.npy
2025-03-29 22:37:49 [INFO] Removed cached tracklet feature file (use_cache: False): c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\20\features.npy
2025-03-29 22:37:49 [INFO] Determine soccer balls in image(s) using pre-trained model.
2025-03-29 22:37:49 [INFO] Removed cached tracklet feature file (use_cache: False): c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\21\features.npy
2025-03-29 2

Processing Batch Tracklets (0-32):   0%|          | 0/32 [00:00<?, ?it/s]

2025-03-29 22:37:50 [INFO] Found 0 balls, Ball list: []
2025-03-29 22:37:50 [INFO] Found 0 balls, Ball list: []
2025-03-29 22:37:50 [INFO] Found 0 balls, Ball list: []
2025-03-29 22:37:50 [INFO] Found 0 balls, Ball list: []
2025-03-29 22:37:50 [INFO] Found 0 balls, Ball list: []
2025-03-29 22:37:50 [INFO] Found 0 balls, Ball list: []
2025-03-29 22:37:50 [INFO] Found 0 balls, Ball list: []
2025-03-29 22:37:50 [INFO] Found 0 balls, Ball list: []


c:\Users\colin\miniconda3\envs\UBC\Lib\site-packages\pytorch_lightning\utilities\migration\migration.py:208: You have multiple `ModelCheckpoint` callback states in this checkpoint, but we found state keys that would end up colliding with each other after an upgrade, which means we can't differentiate which of your checkpoint callbacks needs which states. At least one of your `ModelCheckpoint` callbacks will not be able to reload the state.
Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:37:52 [INFO] Saved features for tracklet with shape (311, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\4\features.npy
2025-03-29 22:37:52 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:37:54 [INFO] Saved features for tracklet with shape (145, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\21\features.npy
2025-03-29 22:37:54 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:37:55 [INFO] Saved features for tracklet with shape (461, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\3\features.npy
2025-03-29 22:37:55 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:37:57 [INFO] Saved features for tracklet with shape (124, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\24\features.npy
2025-03-29 22:37:57 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:37:58 [INFO] Saved features for tracklet with shape (132, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\20\features.npy
2025-03-29 22:37:58 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:38:00 [INFO] Saved features for tracklet with shape (144, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\22\features.npy
2025-03-29 22:38:00 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:38:02 [INFO] Saved features for tracklet with shape (131, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\23\features.npy
2025-03-29 22:38:02 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:38:03 [INFO] 2025-03-29 22:38:02 [INFO] Threshold for scores: 0.999, rounds: 3
2025-03-29 22:38:02 [INFO] Round 1: Excluded 3 outliers
2025-03-29 22:38:02 [INFO] Round 1: Average score: 0.9951072931289673
2025-03-29 22:38:02 [INFO] Excluded images: ['4_97.jpg', '4_98.jpg', '4_99.jpg']
2025-03-29 22:38:02 [INFO] Round 2: Excluded 3 outliers
2025-03-29 22:38:02 [INFO] Round 2: Average score: 0.999954104423523
2025-03-29 22:38:02 [INFO] Excluded images: ['4_97.jpg', '4_98.jpg', '4_99.jpg']
2025-03-29 22:38:02 [INFO] Round 3: Excluded 3 outliers
2025-03-29 22:38:02 [INFO] Round 3: Average score: 0.999954104423523
2025-03-29 22:38:02 [INFO] Excluded images: ['4_97.jpg', '4_98.jpg', '4_99.jpg']

2025-03-29 22:38:03 [ERROR] 
2025-03-29 22:38:03 [INFO] Done removing outliers
2025-03-29 22:38:03 [INFO] Running model chain on preprocessed image(s).
2025-03-29 22:38:03 [INFO] Classifying legibility of image(s) using pre-trained model.


c:\Users\colin\miniconda3\envs\UBC\Lib\site-packages\torchvision\models\_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
c:\Users\colin\miniconda3\envs\UBC\Lib\site-packages\torchvision\models\_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=ResNet34_Weights.IMAGENET1K_V1`. You can also use `weights=ResNet34_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)


2025-03-29 22:38:05 [INFO] Saved features for tracklet with shape (123, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\25\features.npy
2025-03-29 22:38:05 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file
2025-03-29 22:38:06 [INFO] 2025-03-29 22:38:04 [INFO] Threshold for scores: 0.999, rounds: 3
2025-03-29 22:38:04 [INFO] Round 1: Excluded 1 outliers
2025-03-29 22:38:04 [INFO] Round 1: Average score: 0.9964889287948608
2025-03-29 22:38:04 [INFO] Excluded images: ['21_750.jpg']
2025-03-29 22:38:04 [INFO] Round 2: Excluded 1 outliers
2025-03-29 22:38:04 [INFO] Round 2: Average score: 0.999954104423523
2025-03-29 22:38:04 [INFO] Excluded images: ['21_750.jpg']
2025-03-29 22:38:04 [INFO] Round 3: Excluded 1 outliers
2025-03-29 22:38:04 [INFO] Round 3: Average score: 0.999954104423523
2025-03-29 22:38:04 [INFO] Excluded images: ['21_750.jpg']

2025-03-29 22:38:06 [ERROR] 
2025-03-29 22

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:38:20 [INFO] Outputs (thresh=0.7): [[0.5695058107376099], [0.5960512161254883], [0.5634027123451233], [0.5664646029472351]]
2025-03-29 22:38:20 [INFO] Outputs (thresh=0.7): [[0.5707567930221558], [0.5877200365066528], [0.5730789303779602], [0.5719064474105835]]
2025-03-29 22:38:20 [INFO] Outputs (thresh=0.7): [[0.5848327279090881], [0.5671901106834412], [0.5675296187400818], [0.5548633933067322]]
2025-03-29 22:38:20 [INFO] Outputs (thresh=0.7): [[0.5561761260032654], [0.5726381540298462], [0.5558911561965942], [0.5542927980422974]]
2025-03-29 22:38:20 [INFO] Outputs (thresh=0.7): [[0.564801812171936], [0.5425273180007935], [0.5618180632591248], [0.5398374199867249]]
2025-03-29 22:38:21 [INFO] Outputs (thresh=0.7): [[0.5519108772277832], [0.5619466304779053], [0.5581592321395874], [0.5471945405006409]]
2025-03-29 22:38:21 [INFO] Outputs (thresh=0.7): [[0.529600203037262], [0.5469427108764648], [0.5492674708366394], [0.5439652800559998]]
2025-03-29 22:38:21 [INFO] Outputs (

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:38:25 [INFO] Outputs (thresh=0.7): [[0.5389267206192017], [0.5359964966773987], [0.545661449432373], [0.5447208285331726]]
2025-03-29 22:38:25 [INFO] Outputs (thresh=0.7): [[0.5475783348083496], [0.5571233034133911], [0.5469833612442017], [0.5473675727844238]]
2025-03-29 22:38:25 [INFO] Outputs (thresh=0.7): [[0.5474059581756592], [0.5398448705673218], [0.5468218326568604], [0.5463885068893433]]
2025-03-29 22:38:25 [INFO] Outputs (thresh=0.7): [[0.5461967587471008], [0.546536922454834], [0.5464145541191101], [0.5492523908615112]]
2025-03-29 22:38:25 [INFO] Outputs (thresh=0.7): [[0.5468020439147949], [0.5453812479972839], [0.5427525639533997], [0.5411219596862793]]
2025-03-29 22:38:25 [INFO] Outputs (thresh=0.7): [[0.5431455373764038], [0.5462756156921387], [0.5483824014663696], [0.5454514622688293]]
2025-03-29 22:38:25 [INFO] Outputs (thresh=0.7): [[0.54766446352005], [0.5429422855377197], [0.5499741435050964], [0.5539368391036987]]
2025-03-29 22:38:25 [INFO] Outputs (th

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:38:33 [INFO] Tracklet 24 is legible.
2025-03-29 22:38:33 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\24\legible_results.json
2025-03-29 22:38:33 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\24\legible_results.json
2025-03-29 22:38:33 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\24\illegible_results.json
2025-03-29 22:38:33 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\24\illegible_results.json
2025-03-29 22:38:33 [INFO] Legibility classification complete.
2025-03-29 22:38:33 [INFO] Removed cached tracklet feature file (use_cache: False): c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognitio

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:38:36 [INFO] Outputs (thresh=0.7): [[0.6289012432098389], [0.6125519275665283], [0.6132603287696838], [0.6302050352096558]]
2025-03-29 22:38:36 [INFO] Outputs (thresh=0.7): [[0.6333698630332947], [0.6101424098014832], [0.6212921142578125], [0.6161680817604065]]
2025-03-29 22:38:36 [INFO] Outputs (thresh=0.7): [[0.6312100291252136], [0.626392126083374], [0.6096027493476868], [0.6161601543426514]]
2025-03-29 22:38:36 [INFO] Outputs (thresh=0.7): [[0.6214651465415955], [0.6178852319717407], [0.6321055889129639], [0.6303505897521973]]
2025-03-29 22:38:36 [INFO] Outputs (thresh=0.7): [[0.6285197734832764], [0.6291900873184204], [0.6232433915138245], [0.6304513812065125]]
2025-03-29 22:38:36 [INFO] Outputs (thresh=0.7): [[0.6175352334976196], [0.6337250471115112], [0.6376532912254333], [0.6330816745758057]]
2025-03-29 22:38:36 [INFO] Outputs (thresh=0.7): [[0.6172363758087158], [0.6334688663482666], [0.6181038022041321], [0.6219256520271301]]
2025-03-29 22:38:36 [INFO] Outputs 

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:38:39 [INFO] Tracklet 22 is illegible.
2025-03-29 22:38:39 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\22\legible_results.json
2025-03-29 22:38:39 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\22\legible_results.json
2025-03-29 22:38:39 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\22\illegible_results.json
2025-03-29 22:38:39 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\22\illegible_results.json
2025-03-29 22:38:39 [INFO] Legibility classification complete.
2025-03-29 22:38:39 [INFO] Processed tracklet: 22
2025-03-29 22:38:39 [INFO] Removed cached tracklet feature file (use_cache: False): c:\U

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:38:41 [INFO] Outputs (thresh=0.7): [[0.6280949115753174], [0.6306779384613037]]
2025-03-29 22:38:42 [INFO] Tracklet 23 is illegible.
2025-03-29 22:38:42 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\23\legible_results.json
2025-03-29 22:38:42 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\23\legible_results.json
2025-03-29 22:38:42 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\23\illegible_results.json
2025-03-29 22:38:42 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\23\illegible_results.json
2025-03-29 22:38:42 [INFO] Legibility classification complete.
2025-03-29 22:38:42 [INFO] Removed cached tr

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:38:45 [INFO] 2025-03-29 22:38:44 [INFO] Threshold for scores: 0.999, rounds: 3
2025-03-29 22:38:44 [INFO] Round 1: Excluded 0 outliers
2025-03-29 22:38:44 [INFO] Round 1: Average score: 0.9999541640281677
2025-03-29 22:38:44 [INFO] Excluded images: []
2025-03-29 22:38:44 [INFO] Round 2: Excluded 0 outliers
2025-03-29 22:38:44 [INFO] Round 2: Average score: 0.9999541640281677
2025-03-29 22:38:44 [INFO] Excluded images: []
2025-03-29 22:38:44 [INFO] Round 3: Excluded 0 outliers
2025-03-29 22:38:44 [INFO] Round 3: Average score: 0.9999541640281677
2025-03-29 22:38:44 [INFO] Excluded images: []

2025-03-29 22:38:45 [ERROR] 
2025-03-29 22:38:45 [INFO] Done removing outliers
2025-03-29 22:38:45 [INFO] Running model chain on preprocessed image(s).
2025-03-29 22:38:45 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-29 22:38:46 [INFO] Saved features for tracklet with shape (36, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\Socce

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:38:48 [INFO] Saved features for tracklet with shape (91, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\50\features.npy
2025-03-29 22:38:48 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file
2025-03-29 22:38:49 [INFO] 2025-03-29 22:38:48 [INFO] Threshold for scores: 0.999, rounds: 3
2025-03-29 22:38:48 [INFO] Round 1: Excluded 0 outliers
2025-03-29 22:38:48 [INFO] Round 1: Average score: 0.999954104423523
2025-03-29 22:38:48 [INFO] Excluded images: []
2025-03-29 22:38:48 [INFO] Round 2: Excluded 0 outliers
2025-03-29 22:38:48 [INFO] Round 2: Average score: 0.999954104423523
2025-03-29 22:38:48 [INFO] Excluded images: []
2025-03-29 22:38:48 [INFO] Round 3: Excluded 0 outliers
2025-03-29 22:38:48 [INFO] Round 3: Average score: 0.999954104423523
2025-03-29 22:38:48 [INFO] Excluded images: []

2025-03-29 22:38:49 [ERROR] 
2025-03-29 22:38:49 [INFO] Done removing outliers
2

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:39:02 [INFO] Outputs (thresh=0.7): [[0.6915891766548157], [0.6797025799751282], [0.6949355006217957], [0.7008433938026428]]
2025-03-29 22:39:03 [INFO] Outputs (thresh=0.7): [[0.6697610020637512], [0.6822656989097595], [0.6889448761940002], [0.6936841607093811]]
2025-03-29 22:39:03 [INFO] Outputs (thresh=0.7): [[0.6882038712501526], [0.6999894380569458], [0.7047685980796814], [0.6844369769096375]]
2025-03-29 22:39:03 [INFO] 2025-03-29 22:39:01 [INFO] Threshold for scores: 0.999, rounds: 3
2025-03-29 22:39:01 [INFO] Round 1: Excluded 1 outliers
2025-03-29 22:39:01 [INFO] Round 1: Average score: 0.9986756443977356
2025-03-29 22:39:01 [INFO] Excluded images: ['47_559.jpg']
2025-03-29 22:39:01 [INFO] Round 2: Excluded 1 outliers
2025-03-29 22:39:01 [INFO] Round 2: Average score: 0.9999541640281677
2025-03-29 22:39:01 [INFO] Excluded images: ['47_559.jpg']
2025-03-29 22:39:01 [INFO] Round 3: Excluded 1 outliers
2025-03-29 22:39:01 [INFO] Round 3: Average score: 0.99995416402816

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:39:06 [INFO] Outputs (thresh=0.7): [[0.5469796061515808], [0.5452314019203186], [0.5457168221473694], [0.5469074845314026]]
2025-03-29 22:39:07 [INFO] 2025-03-29 22:39:05 [INFO] Threshold for scores: 0.999, rounds: 3
2025-03-29 22:39:05 [INFO] Round 1: Excluded 0 outliers
2025-03-29 22:39:05 [INFO] Round 1: Average score: 0.999954104423523
2025-03-29 22:39:05 [INFO] Excluded images: []
2025-03-29 22:39:05 [INFO] Round 2: Excluded 0 outliers
2025-03-29 22:39:05 [INFO] Round 2: Average score: 0.999954104423523
2025-03-29 22:39:05 [INFO] Excluded images: []
2025-03-29 22:39:05 [INFO] Round 3: Excluded 0 outliers
2025-03-29 22:39:05 [INFO] Round 3: Average score: 0.999954104423523
2025-03-29 22:39:05 [INFO] Excluded images: []

2025-03-29 22:39:07 [ERROR] 
2025-03-29 22:39:07 [INFO] Done removing outliers
2025-03-29 22:39:07 [INFO] Running model chain on preprocessed image(s).
2025-03-29 22:39:07 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-29 22

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:39:11 [INFO] Outputs (thresh=0.7): [[0.6122949719429016], [0.5929642915725708]]
2025-03-29 22:39:12 [INFO] Outputs (thresh=0.7): [[0.649163544178009], [0.6499212980270386], [0.6435308456420898], [0.650482177734375]]
2025-03-29 22:39:12 [INFO] Outputs (thresh=0.7): [[0.6397124528884888], [0.6664276123046875], [0.6535272002220154], [0.6347967386245728]]
2025-03-29 22:39:12 [INFO] Outputs (thresh=0.7): [[0.6456976532936096], [0.6620967388153076], [0.6377072930335999], [0.6460217833518982]]
2025-03-29 22:39:12 [INFO] Outputs (thresh=0.7): [[0.5998663902282715], [0.613184928894043], [0.6125265955924988], [0.6128248572349548]]
2025-03-29 22:39:12 [INFO] Outputs (thresh=0.7): [[0.6080150604248047], [0.6534119844436646], [0.6526311039924622], [0.63563472032547]]
2025-03-29 22:39:12 [INFO] Outputs (thresh=0.7): [[0.64718097448349], [0.6417437195777893], [0.6443548202514648], [0.6557494401931763]]
2025-03-29 22:39:12 [INFO] Tracklet 40 is illegible.
2025-03-29 22:39:13 [INFO] Savin

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:39:18 [INFO] Outputs (thresh=0.7): [[0.5981273651123047], [0.5870873332023621], [0.5995669960975647], [0.591168999671936]]
2025-03-29 22:39:18 [INFO] Outputs (thresh=0.7): [[0.6223257780075073], [0.6105587482452393], [0.624467670917511], [0.6089897155761719]]
2025-03-29 22:39:18 [INFO] Outputs (thresh=0.7): [[0.6095921397209167], [0.6281442642211914], [0.6005013585090637], [0.621516764163971]]
2025-03-29 22:39:18 [INFO] Outputs (thresh=0.7): [[0.6364532709121704], [0.6245066523551941], [0.6398122906684875], [0.6284893155097961]]
2025-03-29 22:39:18 [INFO] Outputs (thresh=0.7): [[0.6145666837692261], [0.6201955676078796], [0.6377145648002625], [0.6181737780570984]]
2025-03-29 22:39:18 [INFO] Outputs (thresh=0.7): [[0.6259803175926208], [0.6186115741729736], [0.6012147068977356], [0.6042052507400513]]
2025-03-29 22:39:18 [INFO] Outputs (thresh=0.7): [[0.6294906735420227], [0.618564248085022], [0.624098539352417], [0.6652706861495972]]
2025-03-29 22:39:18 [INFO] Outputs (thr

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:39:22 [INFO] Outputs (thresh=0.7): [[0.606388509273529], [0.609664797782898], [0.6157395839691162], [0.6243464946746826]]
2025-03-29 22:39:22 [INFO] Outputs (thresh=0.7): [[0.6119632124900818], [0.6154466271400452], [0.6127991676330566], [0.6210755109786987]]
2025-03-29 22:39:22 [INFO] Outputs (thresh=0.7): [[0.6286389827728271], [0.6342562437057495], [0.6042474508285522], [0.6261738538742065]]
2025-03-29 22:39:22 [INFO] Tracklet 49 is illegible.
2025-03-29 22:39:22 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\49\legible_results.json
2025-03-29 22:39:22 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\49\legible_results.json
2025-03-29 22:39:22 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\49\

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:39:24 [INFO] 2025-03-29 22:39:23 [INFO] Threshold for scores: 0.999, rounds: 3
2025-03-29 22:39:23 [INFO] Round 1: Excluded 0 outliers
2025-03-29 22:39:23 [INFO] Round 1: Average score: 0.9999541640281677
2025-03-29 22:39:23 [INFO] Excluded images: []
2025-03-29 22:39:23 [INFO] Round 2: Excluded 0 outliers
2025-03-29 22:39:23 [INFO] Round 2: Average score: 0.9999541640281677
2025-03-29 22:39:23 [INFO] Excluded images: []
2025-03-29 22:39:23 [INFO] Round 3: Excluded 0 outliers
2025-03-29 22:39:23 [INFO] Round 3: Average score: 0.9999541640281677
2025-03-29 22:39:23 [INFO] Excluded images: []

2025-03-29 22:39:24 [ERROR] 
2025-03-29 22:39:24 [INFO] Done removing outliers
2025-03-29 22:39:24 [INFO] Running model chain on preprocessed image(s).
2025-03-29 22:39:24 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-29 22:39:25 [INFO] Tracklet 50 is illegible.
2025-03-29 22:39:25 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:39:27 [INFO] Saved features for tracklet with shape (54, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\76\features.npy
2025-03-29 22:39:27 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:39:28 [INFO] 2025-03-29 22:39:27 [INFO] Threshold for scores: 0.999, rounds: 3
2025-03-29 22:39:27 [INFO] Round 1: Excluded 0 outliers
2025-03-29 22:39:27 [INFO] Round 1: Average score: 0.9999541640281677
2025-03-29 22:39:27 [INFO] Excluded images: []
2025-03-29 22:39:27 [INFO] Round 2: Excluded 0 outliers
2025-03-29 22:39:27 [INFO] Round 2: Average score: 0.9999541640281677
2025-03-29 22:39:27 [INFO] Excluded images: []
2025-03-29 22:39:27 [INFO] Round 3: Excluded 0 outliers
2025-03-29 22:39:27 [INFO] Round 3: Average score: 0.9999541640281677
2025-03-29 22:39:27 [INFO] Excluded images: []

2025-03-29 22:39:28 [ERROR] 
2025-03-29 22:39:28 [INFO] Done removing outliers
2025-03-29 22:39:28 [INFO] Running model chain on preprocessed image(s).
2025-03-29 22:39:28 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-29 22:39:29 [INFO] Saved features for tracklet with shape (53, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\Socce

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:39:36 [INFO] 2025-03-29 22:39:35 [INFO] Threshold for scores: 0.999, rounds: 3
2025-03-29 22:39:35 [INFO] Round 1: Excluded 3 outliers
2025-03-29 22:39:35 [INFO] Round 1: Average score: 0.9977883696556091
2025-03-29 22:39:35 [INFO] Excluded images: ['71_97.jpg', '71_98.jpg', '71_99.jpg']
2025-03-29 22:39:35 [INFO] Round 2: Excluded 3 outliers
2025-03-29 22:39:35 [INFO] Round 2: Average score: 0.9999542236328125
2025-03-29 22:39:35 [INFO] Excluded images: ['71_97.jpg', '71_98.jpg', '71_99.jpg']
2025-03-29 22:39:35 [INFO] Round 3: Excluded 3 outliers
2025-03-29 22:39:35 [INFO] Round 3: Average score: 0.9999542236328125
2025-03-29 22:39:35 [INFO] Excluded images: ['71_97.jpg', '71_98.jpg', '71_99.jpg']

2025-03-29 22:39:36 [ERROR] 
2025-03-29 22:39:36 [INFO] Done removing outliers
2025-03-29 22:39:36 [INFO] Running model chain on preprocessed image(s).
2025-03-29 22:39:36 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-29 22:39:38 [INFO] Saved feat

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:39:41 [INFO] Saved features for tracklet with shape (36, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\81\features.npy
2025-03-29 22:39:41 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file
2025-03-29 22:39:41 [INFO] 2025-03-29 22:39:40 [INFO] Threshold for scores: 0.999, rounds: 3
2025-03-29 22:39:40 [INFO] Round 1: Excluded 0 outliers
2025-03-29 22:39:40 [INFO] Round 1: Average score: 0.9999541640281677
2025-03-29 22:39:40 [INFO] Excluded images: []
2025-03-29 22:39:40 [INFO] Round 2: Excluded 0 outliers
2025-03-29 22:39:40 [INFO] Round 2: Average score: 0.9999541640281677
2025-03-29 22:39:40 [INFO] Excluded images: []
2025-03-29 22:39:40 [INFO] Round 3: Excluded 0 outliers
2025-03-29 22:39:40 [INFO] Round 3: Average score: 0.9999541640281677
2025-03-29 22:39:40 [INFO] Excluded images: []

2025-03-29 22:39:41 [ERROR] 
2025-03-29 22:39:41 [INFO] Done removing outlier

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:39:54 [INFO] Outputs (thresh=0.7): [[0.6025952100753784], [0.6248307824134827], [0.6192153096199036], [0.6234626173973083]]
2025-03-29 22:39:54 [INFO] Outputs (thresh=0.7): [[0.6285797357559204], [0.6151406764984131], [0.618332028388977], [0.620781660079956]]
2025-03-29 22:39:54 [INFO] Outputs (thresh=0.7): [[0.6255852580070496], [0.6169458031654358], [0.600871205329895], [0.6145573258399963]]
2025-03-29 22:39:54 [INFO] Outputs (thresh=0.7): [[0.6266047358512878], [0.6165692210197449], [0.631443977355957], [0.6365764737129211]]
2025-03-29 22:39:54 [INFO] Outputs (thresh=0.7): [[0.5991686582565308], [0.6293766498565674], [0.5977596640586853], [0.5964124798774719]]
2025-03-29 22:39:54 [INFO] Outputs (thresh=0.7): [[0.6037736535072327], [0.5933135747909546], [0.5822415351867676], [0.5864501595497131]]
2025-03-29 22:39:54 [INFO] Outputs (thresh=0.7): [[0.5863444209098816], [0.6080911159515381], [0.6070302724838257], [0.6038036942481995]]
2025-03-29 22:39:54 [INFO] Outputs (th

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:39:56 [INFO] Outputs (thresh=0.7): [[0.6533452272415161], [0.6365419030189514], [0.6372209787368774], [0.6501198410987854]]
2025-03-29 22:39:56 [INFO] Outputs (thresh=0.7): [[0.6503392457962036], [0.6208699345588684], [0.6561539173126221], [0.6435240507125854]]
2025-03-29 22:39:56 [INFO] Outputs (thresh=0.7): [[0.6374421119689941], [0.6476388573646545], [0.6430554389953613], [0.6548078060150146]]
2025-03-29 22:39:56 [INFO] 2025-03-29 22:39:55 [INFO] Threshold for scores: 0.999, rounds: 3
2025-03-29 22:39:55 [INFO] Round 1: Excluded 0 outliers
2025-03-29 22:39:55 [INFO] Round 1: Average score: 0.9999541640281677
2025-03-29 22:39:55 [INFO] Excluded images: []
2025-03-29 22:39:55 [INFO] Round 2: Excluded 0 outliers
2025-03-29 22:39:55 [INFO] Round 2: Average score: 0.9999541640281677
2025-03-29 22:39:55 [INFO] Excluded images: []
2025-03-29 22:39:55 [INFO] Round 3: Excluded 0 outliers
2025-03-29 22:39:55 [INFO] Round 3: Average score: 0.9999541640281677
2025-03-29 22:39:55 [

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:40:01 [INFO] 2025-03-29 22:39:59 [INFO] Threshold for scores: 0.999, rounds: 3
2025-03-29 22:39:59 [INFO] Round 1: Excluded 0 outliers
2025-03-29 22:39:59 [INFO] Round 1: Average score: 0.999954104423523
2025-03-29 22:39:59 [INFO] Excluded images: []
2025-03-29 22:39:59 [INFO] Round 2: Excluded 0 outliers
2025-03-29 22:39:59 [INFO] Round 2: Average score: 0.999954104423523
2025-03-29 22:39:59 [INFO] Excluded images: []
2025-03-29 22:39:59 [INFO] Round 3: Excluded 0 outliers
2025-03-29 22:39:59 [INFO] Round 3: Average score: 0.999954104423523
2025-03-29 22:39:59 [INFO] Excluded images: []

2025-03-29 22:40:01 [ERROR] 
2025-03-29 22:40:01 [INFO] Done removing outliers
2025-03-29 22:40:01 [INFO] Running model chain on preprocessed image(s).
2025-03-29 22:40:01 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-29 22:40:03 [INFO] Saved features for tracklet with shape (322, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerN

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:40:06 [INFO] Saved features for tracklet with shape (80, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\89\features.npy
2025-03-29 22:40:06 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file
2025-03-29 22:40:08 [INFO] Outputs (thresh=0.7): [[0.6588032841682434], [0.6353539824485779], [0.6488216519355774], [0.635586142539978]]
2025-03-29 22:40:08 [INFO] Outputs (thresh=0.7): [[0.6543231010437012], [0.6449874043464661], [0.6521527171134949], [0.6469012498855591]]
2025-03-29 22:40:08 [INFO] Outputs (thresh=0.7): [[0.6295899152755737], [0.6142917275428772], [0.6351362466812134], [0.6302425265312195]]
2025-03-29 22:40:08 [INFO] Outputs (thresh=0.7): [[0.5998561382293701], [0.6089940071105957], [0.5968496799468994], [0.5839149951934814]]
2025-03-29 22:40:08 [INFO] Outputs (thresh=0.7): [[0.5821461081504822], [0.5867263078689575], [0.5908837914466858], [0.6085456609725952]]
2

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:40:10 [INFO] Saved features for tracklet with shape (5, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\90\features.npy
2025-03-29 22:40:10 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file
2025-03-29 22:40:11 [INFO] Outputs (thresh=0.7): [[0.6461894512176514], [0.6162492632865906], [0.6459754109382629], [0.6479713916778564]]
2025-03-29 22:40:11 [INFO] Outputs (thresh=0.7): [[0.576312780380249], [0.5793508887290955], [0.5900842547416687], [0.5914056897163391]]
2025-03-29 22:40:11 [INFO] Outputs (thresh=0.7): [[0.6438031792640686], [0.6409425735473633], [0.6332560181617737], [0.6535278558731079]]
2025-03-29 22:40:12 [INFO] Outputs (thresh=0.7): [[0.634009599685669], [0.6439292430877686], [0.6439253091812134], [0.6494289636611938]]
2025-03-29 22:40:12 [INFO] Outputs (thresh=0.7): [[0.6424540281295776], [0.6432241201400757], [0.6364737153053284], [0.6358429789543152]]
202

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-29 22:40:14 [INFO] Outputs (thresh=0.7): [[0.6341527700424194], [0.6253325343132019], [0.6441253423690796], [0.6556292176246643]]
2025-03-29 22:40:14 [INFO] Outputs (thresh=0.7): [[0.6342112421989441], [0.6335719227790833], [0.6449366211891174], [0.6507388353347778]]
2025-03-29 22:40:14 [INFO] Outputs (thresh=0.7): [[0.6563124656677246], [0.6665931344032288], [0.6573025584220886], [0.6587967276573181]]
2025-03-29 22:40:14 [INFO] Outputs (thresh=0.7): [[0.6638843417167664], [0.6489195823669434], [0.654345691204071], [0.6397898197174072]]
2025-03-29 22:40:14 [INFO] Outputs (thresh=0.7): [[0.6526869535446167], [0.6314519643783569], [0.6464203000068665], [0.6495451331138611]]
2025-03-29 22:40:14 [INFO] Outputs (thresh=0.7): [[0.6520093083381653], [0.6438151597976685], [0.6449438333511353], [0.657642662525177]]
2025-03-29 22:40:14 [INFO] Outputs (thresh=0.7): [[0.6505154371261597], [0.6538758873939514], [0.6261710524559021], [0.6259454488754272]]
2025-03-29 22:40:14 [INFO] Outputs (

KeyboardInterrupt: 

## Benchmark (current pipeline metrics)
### Results
- 2025-03-29 14:44:05 [INFO] Correct 1051 out of 1210. Accuracy 86.85950413223141%.
- 2025-03-29 14:44:05 [INFO] TP=766, TN=285, FP=70, FN=89
- 2025-03-29 14:44:05 [INFO] Precision=0.916267942583732, Recall=0.895906432748538
- 2025-03-29 14:44:05 [INFO] F1=0.9059727971614429
- 2025-03-29 14:44:05 [INFO] Saved failed legibility cases to:  c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\common_data\failed_legibility_cases.json

### Analysis
legible_but_marked_illegible: 89 cases
109, 88, 34, 151, 19, 68, 67, 128, 57, 178, 189, 196, 214, 242, 258, 271, 304, 327, 320, 352, 375, 378, 414, 416, 439, 425, 451, 453, 454, 466, 499, 511, 513, 526, 528, 567, 576, 594, 619, 623, 646, 687, 682, 709, 711, 717, 739, 758, 767, 795, 788, 840, 843, 832, 841, 852, 847, 851, 856, 850, 868, 866, 871, 901, 902, 904, 900, 907, 928, 939, 944, 961, 984, 983, 1005, 1039, 1064, 1063, 1077, 1091, 1101, 1137, 1152, 1153, 1160, 1167, 1157, 1165, 1206

illegible_but_marked_legible: 70 cases
168, 167, 162, 97, 39, 87, 89, 134, 61, 172, 164, 199, 215, 260, 267, 269, 281, 276, 288, 295, 317, 325, 340, 345, 374, 394, 388, 386, 391, 406, 408, 434, 437, 459, 471, 514, 532, 558, 683, 748, 770, 790, 802, 827, 861, 869, 879, 897, 895, 909, 964, 981, 999, 1023, 1031, 1025, 1033, 1030, 1055, 1088, 1087, 1169, 1175, 1174, 1173, 1183, 1170, 1179, 1184, 1185

- Total misclassifications: 159
- False negatives (legible marked as illegible): 89
- False positives (illegible marked as legible): 70
- Percentage of false negatives: 7.36%
- Percentage of false positives: 5.79%
- Combined percentage of errors as a percentage of total cases: 13.14%

In [ ]:
import json
failed_cases = f"{DataPaths.COMMON_PROCESSED_OUTPUT_DATA_TEST.value}/failed_legibility_cases.json"

# Load this JSON file
with open(failed_cases, 'r') as f:
    failed_cases_dict = json.load(f)
    # Print the every key and its associated value (value will be a list so need to make that printable with ",".join):
    for key, value in failed_cases_dict.items():
        print(f"{key}: {len(value)} cases")
        if len(value) > 0:
            print(f"{', '.join(value)}")
        print()
    
    # Summary statistics
    legible_but_marked_illegible = len(failed_cases_dict.get("legible_but_marked_illegible", []))
    illegible_but_marked_legible = len(failed_cases_dict.get("illegible_but_marked_legible", []))
    total_errors = legible_but_marked_illegible + illegible_but_marked_legible
    
    print(f"Total misclassifications: {total_errors}")
    print(f"False negatives (legible marked as illegible): {legible_but_marked_illegible}")
    print(f"False positives (illegible marked as legible): {illegible_but_marked_legible}")
    print(f"Percentage of false negatives: {legible_but_marked_illegible / 1210 * 100:.2f}%")
    print(f"Percentage of false positives: {illegible_but_marked_legible / 1210 * 100:.2f}%")
    print(f"Combined percentage of errors as a percentage of total cases: {total_errors / 1210 * 100:.2f}%")

In [ ]:
# import timm
# import torch
# from torch import nn

# model = timm.create_model("timm/vit_base_patch16_224.orig_in21k_ft_in1k", pretrained=False)
# model.head = nn.Linear(model.head.in_features, 10)
# model.load_state_dict(
#     torch.hub.load_state_dict_from_url(
#         "https://huggingface.co/edadaltocg/vit_base_patch16_224_in21k_ft_svhn/resolve/main/pytorch_model.bin",
#         map_location="cpu",
#         file_name="vit_base_patch16_224_in21k_ft_svhn.pth",
#     )
# )